src/cargo/util/toml/mod.rs

```diff
@@ -1944,7 +1965,17 @@ fn detailed_dep_to_dependency<P: ResolveToPath + Clone>(
     }
 
     if let Some(p) = orig.public {
-        cx.features.require(Feature::public_dependency())?;
```

Comment:
Brought back [feedback from t-release](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/-Zpublic-dependency.20syntax.20change/near/418323741) for <https://github.com/rust-lang/cargo/pull/13340#issuecomment-1907284634>. We might want to support `cargo-features = ["public-dependency"]` for at least one release cycle, so it won't break self-rebuild. Hence this should be kept I believe (and probably other places `cli_unstable().public_dependency` gate are used as well).

Would be good if we have a test verifying it won't bail out or warn in this case.

Reply:
I had addded a new method named `is_public_dependency_enabled` to support ``support cargo-features = ["public-dependency"].

New test cases had addded to `pub_priv.rs`.  (there is one test for the mixture of -Z and cargo-features.)